geode-web/src/main/java/org/apache/geode/management/internal/web/controllers/ShellCommandsController.java

```diff
@@ -148,6 +149,11 @@
     parameters = (parameters != null ? parameters : ArrayUtils.EMPTY_OBJECT_ARRAY);
     MBeanServer mBeanServer = getMBeanServer();
     ObjectName objectName = ObjectName.getInstance(decode(resourceName));
+    // GEODE-6182
+    for (int i = 0; i < signature.length; i++) {
+      parameters[i] = GeodeConverter.convertToActualType(parameters[i].toString(), signature[i]);
```

Comment:
is looks like the domain of all possible valid `signature` strings is anything `getClass().getName()` can return...how confident are we that Geode never passes any parameters other than primitives?   https://docs.oracle.com/javase/7/docs/api/javax/management/MBeanServer.html#invoke(javax.management.ObjectName,%20java.lang.String,%20java.lang.Object[],%20java.lang.String[])

Reply:
Note spring already turned everything into String already, whatever conversion we did here, is already an "improvement". Also, I believe to invoke jmx operations, only primitive arguments are supported in jmx calls.

Reply:
Besides primitives, I think String would be a likely type.

`convertToActualType` will not even handle String correctly, since `String.class.getName()` is `java.lang.String` not `string`.  Maybe we should at least have a test for String in addition to int and bool.

Reply:
resolved by latest commit, looks good now